src/lib/api.ts

```diff
@@ -1046,6 +1046,41 @@ export const searchParentContent = cache(
     },
 );
 
+/**
+ * Search content in a Site or specific SiteSpaces.
+ */
+export const searchSiteContent = cache(
+    'api.searchSiteContent',
+    async (
+        organizationId: string,
+        siteId: string,
+        query: string,
+        siteSpaceIds: string[] = [],
+        /** A cache bust key to avoid revalidating lot of cache entries by tags */
+        cacheKey?: string,
+        options?: CacheFunctionOptions,
+    ) => {
+        const response = await api().orgs.searchSiteContent(
+            organizationId,
+            siteId,
+            {
+                query,
+                siteSpaceIds,
```

Comment:
I didn't review the API design of this endpoint, but I have mixed feeling at first sight. What does it mean when an empty array is passed? Why not pass `undefined`? Is it a filter?

Reply:
From the API spec, it looks like a filter yeah, so we should simply not pass it when searching across all.

Reply:
yeah, it's a filter. I'll update the API.

Reply:
ah, I already did that in the API as you can pass undefined for the same thing. just didn't do that here. fixing...